public/ledger.js

```diff
@@ -0,0 +1,290 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// Need to require this to use ledger libs
+// https://github.com/LedgerHQ/ledgerjs/issues/332
+require('babel-polyfill');
+const Transport = require('@ledgerhq/hw-transport-node-hid').default;
+const constants = require('./constants');
+
+const maxLedgerBuffer = 255;
+
+class Ledger {
+  
+  static parseLedgerError(e) {
+    if (e.name && e.name === 'TransportStatusError') {
+      switch (e.statusCode) {
+        case 0x6985:
+          return {status: e.statusCode, message: 'Request denied by user on Ledger'};
+        case 0x6b00:  // SW_DEVELOPER_ERROR
+        case 0x6b01:  // SW_INVALID_PARAM
+        case 0x6b02:  // SW_IMPROPER_INIT
+        default:
+          return {status: e.statusCode, message: 'Error communicating with Ledger'};
+      }
+    }
+    return e;
+  }
+
+  constructor() {
+    // Electron main window to send command to renderer process
+    this.mainWindow = null;
+    // Ledger transport
+    this.transport = null;
+    // Timeout for ledger connection
+    this.timeout = 10000;
+    // If hathor app is opened on ledger
+    this.hathorAppOpened = false;
+
+    this.subscriptor = Transport.listen({
+      next: e => {
+        if (e.type === 'add') {
+          // All events called here we check if we are in the hathor app
```

Comment:
I don't get it.

Reply:
The comment is not explaining well. This event is called when any app is opened, bitcoin, ethereum, or any other app on ledger, so we must always check that we are in our app. This is what the `getVersion` method does.

Reply:
Improved the comment

Reply:
Thanks!